pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -154,13 +164,45 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 			return admission.ResponseAllowed(), nil
 		}
 	}
+	// Check for escalations in fleet workspace rules
+	if newGR.InheritedFleetWorkspacePermissions.ResourceRules != nil || newGR.InheritedFleetWorkspacePermissions.WorkspaceVerbs != nil {
+		fleetWorkspaceRules := a.grbResolver.GlobalRoleResolver.FleetWorkspaceRulesFromRole(newGR)
+		workspaceNames, err := a.allFleetWorkspaceNamesExceptLocal()
+		if err != nil {
+			return nil, fmt.Errorf("unable to fecht fleet workspaces: %w", err)
+		}
+		for _, w := range workspaceNames {
+			returnError = errors.Join(returnError, escalateChecker.IsRulesAllowed(fleetWorkspaceRules, a.resolver, w))
```

Comment:
Right now, this seems to be checking if the user has the requested permission in each of the workspaces as a means of escalation checks. 

So for example, let's assume there are 2 fleetworkspaces: `fleet-default`, and `fleet-local`. If I'm trying to edit a global role that has the following permissions:
```yaml
apiVersion: management.cattle.io/v3
kind: GlobalRole
metadata:
  name: example
inheritedFleetWorkspacePermissions:
  resourceRules:
  - apiGroups:
    - ""
    resources: 
    - "pods"
    verbs: 
    -"get"
```
The webhook will check if I have `get` on `pods` in `fleet-default`, which I could already have by virtue of a kubernetes role/binding. The issue is, if someone makes a new workspace `fleet-new` I also gain permissions in that workspace. But the original role/binding didn't give me that new permission - so I've technically escalated my privileges.

In short, currently having permissions in all non-local namespaces is not a gaurantee that you will have them in all future workspaces as well. This issue also applies for the verb check below. 

I would recommend looking at how we do the escalation check for `inheritedClusterRoles` [here](https://github.com/rancher/webhook/blob/109fe534b73bc47374dfd5f8a962d3f3d6545eb6/pkg/resources/management.cattle.io/v3/globalrole/validator.go#L153) and adopting a similar approach.

Reply:
I added the `fleetWorkspaceRules` to the `ClusterRulesFromRole` in the `GlobalRoleResolver`. This approach reuses the existing escalation check logic for the `inheritedClusterRoles`. By reusing the existing logic, the check is now included for both `GlobalRole` and `GlobalRoleBinding`. 
Please have a look and let me know what you think.